cmd/release-controller/main.go

```diff
@@ -156,6 +159,9 @@ func main() {
 
 	flagset.StringVar(&opt.ReleaseArchitecture, "release-architecture", opt.ReleaseArchitecture, "The architecture of the releases to be created (defaults to 'amd64' if not specified).")
 
+	flagset.BoolVar(&opt.PruneGraph, "prune-graph", opt.PruneGraph, "Reads the upgrade graph, prunes edges, and prints the result")
```

Comment:
Would probably prefer a flag where empty = "secret" and you could fold the debug print output onto a different value.  I.e. `--print-pruned-graph=<|secret|debug` where empty string = secret if the flag is set.

Reply:
Ah, you had it prune inline too.  Hrm.  I would probably have two flags - `--prune-graph` that runs and outputs the result and prints how many edges are removed and then `--confirm-prune-graph` that is required to actually prune.

Reply:
I was doing the in-line prune, so that I can get the actual contents of the upgraded graph.  I wasnt planning on adding the save logic until we agreed that the logic was sound and that we liked the results.  That being said, it all exists now..
I've updated the logic as follows:
`./release-controller --release-namespace ocp --job-namespace ci --prune-graph`
Prints out the base64 encoded secret payload
`./release-controller --release-namespace ocp --job-namespace ci --prune-graph --print-pruned-graph debug`
Pretty prints the json output of the graph
`./release-controller --release-namespace ocp --job-namespace ci --prune-graph --confirm-prune-graph`
Will perform the update of the secret